client/src/components/Tool/Buttons/ToolTourGeneratorItem.vue

```diff
@@ -0,0 +1,126 @@
+<script setup lang="ts">
+import { faPuzzlePiece } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
+import { BDropdownItem } from "bootstrap-vue";
+import { storeToRefs } from "pinia";
+import { computed, ref, watch } from "vue";
+
+import { generateTour, type TourDetails } from "@/api/tours";
+import { Toast } from "@/composables/toast";
+import { useHistoryItemsStore } from "@/stores/historyItemsStore";
+import { useHistoryStore } from "@/stores/historyStore";
+import { useTourStore } from "@/stores/tourStore";
+import { errorMessageAsString } from "@/utils/simple-error";
+
+import LoadingSpan from "@/components/LoadingSpan.vue";
+
+const INVALID_UPLOAD_STATES = ["discarded", "failed", "error"];
```

Comment:
Can you use these states instead: https://github.com/galaxyproject/galaxy/blob/dev/client/src/api/jobs.ts#L12